prescient/stats/operations_stats.py

```diff
@@ -274,14 +292,24 @@ def populate_market_settlement(self,
                      (self.observed_virtual_dispatch_levels[g] - self.virtual_gen_cleared_DA[g])*price_RT
                     ) * self.sced_duration_minutes / 60
 
-
-        r_price_DA = self.planning_reserve_price
-        r_price_RT = self.reserve_RT_price
-        self.thermal_reserve_revenue = { g : (self.thermal_reserve_cleared_DA[g]*r_price_DA + \
-                                                ( self.observed_thermal_headroom_levels[g] - \
-                                                   self.thermal_reserve_cleared_DA[g] )*r_price_RT
-                                             ) * self.sced_duration_minutes / 60
-                                         for g in extractor.get_thermal_generators(sced) }
+        self.thermal_per_reserve_revenue = {}
+        self.thermal_total_reserve_revenue = {g : 0.
+                                              for g in extractor.get_thermal_generators(sced)}
+        for res in ruc_market.DA_reserve_requirements.keys():
+            r_price_DA = self.DA_reserve_prices[res]
+            r_price_RT = \
+                self.reserve_RT_prices[res] \
+                if res in self.reserve_RT_prices \
+                else 0.
+            for g in extractor.get_thermal_generators(sced):
+                revenue = \
+                    (self.thermal_reserve_cleared_DA[res,g]*r_price_DA 
+                     + (self.observed_thermal_headroom_levels[g] 
```

Comment:
There should be an equivalent of `thermal_reserve_cleared_RT[res,g]` we can pull out for each product.

Or, if a product exists in the DA but not the RT, we should assume the price to be 0 and ignore this term in the sum. (I don't think we would have the opposite situation.)

Reply:
On more note -- we need to also handle zonal/area-level reserves differently than system-level reserves.

Reply:
This is now implemented as `extractor.get_thermal_reserve_provided(sced, res, g)`.